### PR TITLE
Swagger Documentation Added 

### PR DESCRIPTION
**Pull request by dakshgautam**

### CODE REVIEW

File: squealy/constants.py [hunk 1]
```diff
@@ -6,4 +6,48 @@
...
+            "termsOfService": "http://swagger.io/terms/",
+            "contact": {
+                "email": "apiteam@swagger.io"
+            },
+            "license": {
+                "name": "Apache 2.0",
+                "url": "http://www.apache.org/licenses/LICENSE-2.0.html"
+            }
+        },
+        "host": "localhost:8000",
```
Comment (devashishsharma2302) [inline squealy/constants.py]: host name is hard-coded as localhost. Will this work after deployment?

File: squealy/templates/swagger.html [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
+        break;
+      }
+    }
+  }
+  return cookieValue;
+}
+
+
+  apiCall('http://localhost:8000/swagger-api/',{},'GET');
```
Comment (devashishsharma2302) [inline squealy/templates/swagger.html]: Do not hard-code the host name as localhost. This would fail in production.

File: squealy/views.py [hunk 3]
```diff
@@ -541,3 +558,70 @@ def squealy_interface(request, *args, **kwargs):
...
+                    }
+                }
+            }
+    }
+    return path_content_template
+
+
+@api_view(['GET'])
+def swagger_json_api(request, *args, **kwargs):
+    host = "localhost:8000"
```
Comment (devashishsharma2302) [inline squealy/views.py]: Same hostname issue here as well.

File: squealy/views.py [hunk 4]
```diff
@@ -541,3 +558,70 @@ def squealy_interface(request, *args, **kwargs):
...
+    swagger_json_template = SWAGGER_JSON_TEMPLATE
+    swagger_json_template["host"] = host
+    swagger_dict = SWAGGER_DICT
+    charts_related = ChartsLoaderView.get_charts_swagger(request)
```
Comment (devashishsharma2302) [inline squealy/views.py]: rename to permitted_charts.

File: squealy/views.py [hunk 5]
```diff
@@ -269,6 +275,17 @@ class ChartsLoaderView(APIView):
     authentication_classes = [SessionAuthentication, BasicAuthentication]
     authentication_classes.extend(SquealySettings.get_default_authentication_classes())
 
+    @staticmethod
+    def get_charts_swagger(request):
+        permitted_charts = []
```
Comment (devashishsharma2302) [inline squealy/views.py]: the get_permitted_charts logic is repeated, make a function and reuse in get() and get_charts_swagger().

File: squealy/views.py [hunk 6]
```diff
@@ -541,3 +558,70 @@ def squealy_interface(request, *args, **kwargs):
...
+def make_parameters(param_list):
+    path_content_template = {
+        "get":
+            {
+                "tags": [
+                    "charts"
+                ],
+                "summary": "Charts API",
+                "description": "Add parameters according to the Query",
+                "operationId": "findPetsByStatus",
```
Comment (devashishsharma2302) [inline squealy/views.py]: findPetsByStatus?

File: squealy/templates/swagger.html [hunk 7]
```diff
@@ -0,0 +1,148 @@
+<!-- HTML for static distribution bundle build -->
+{% load staticfiles %}
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Swagger UI</title>
+  <link href="https://fonts.googleapis.com/css?family=Open+Sans:400,700|Source+Code+Pro:300,600|Titillium+Web:400,600,700" rel="stylesheet">
+  <link rel="stylesheet" type="text/css" href="{% static "swagger-ui.css" %}" >
+  <!--<link rel="icon" type="image/png" href="../public/favicon-32x32.png" sizes="32x32" />-->
```
Comment (hemny-singh) [inline squealy/templates/swagger.html]: Remove commented code if not needed.

File: squealy/templates/swagger.html [hunk 8]
```diff
@@ -0,0 +1,148 @@
+<!-- HTML for static distribution bundle build -->
+{% load staticfiles %}
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Swagger UI</title>
+  <link href="https://fonts.googleapis.com/css?family=Open+Sans:400,700|Source+Code+Pro:300,600|Titillium+Web:400,600,700" rel="stylesheet">
```
Comment (hemny-singh) [inline squealy/templates/swagger.html]: Do we need to add Open Sans font family source? Check if we really need to provide source for Open Sans.

File: squealy/templates/swagger.html [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+
+    <symbol viewBox="0 0 24 24" id="expand">
+      <path d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"/>
+    </symbol>
+
+  </defs>
+</svg>
+
+<div id="swagger-ui"></div>
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.0/jquery.min.js"></script>
```
Comment (hemny-singh) [inline squealy/templates/swagger.html]: We already have downloaded jQuery file in `squealyproj/js/static`.

File: squealy/templates/swagger.html [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+<script src="{% static "swagger-ui-standalone-preset.js" %} "> </script>
+<script>
+
+window.foo = function(spec){
+  const ui = SwaggerUIBundle({
+    spec: spec,
+    dom_id: '#swagger-ui',
+    presets: [
+      SwaggerUIBundle.presets.apis,
+      // yay ES6 modules ↘
```
Comment (hemny-singh) [inline squealy/templates/swagger.html]: Add proper comments.

File: squealy/templates/swagger.html [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+      SwaggerUIBundle.plugins.DownloadUrl
+    ],
+    layout: "StandaloneLayout"
+  })
+
+  window.ui = ui
+}
+window.onload = function() {
+
+   function apiCall(uri, data, methodType) {
```
Comment (hemny-singh) [inline squealy/templates/swagger.html]: Indentation is not proper.

File: squealy/constants.py [hunk 12]
```diff
@@ -6,4 +6,48 @@
...
     # Data Manipulation
     'INSERT', 'UPDATE', 'REPLACE', 'DELETE',
-]
\ No newline at end of file
+]
+
+SWAGGER_JSON_TEMPLATE = {
+        "swagger": "2.0",
+        "info": {
+            "description": "This is Squealy Server.  You can find out more about Swagger at [http://swagger.io](http://swagger.io) or on [irc.freenode.net, #swagger](http://swagger.io/irc/).  For this sample, you can use the api key `special-key` to test the authorization filters.",
```
Comment (hemny-singh) [inline squealy/constants.py]: Extra spaces after 'Server.'

File: squealy/constants.py (hunk 1 above)
Comment (dakshgautam) [inline squealy/constants.py]: No

File: squealy/squealy-web/src/Containers/AuthoringInterfaceContainer.js [hunk 13]
```diff
@@ -125,6 +124,8 @@ export default class AuthoringInterfaceContainer extends Component {
       }, () => {
         callback && callback.constructor === Function && callback()
         this.setUrlPath(type)
+        console.log(type,'This is type')
```
Comment (swapnil95) [inline squealy/squealy-web/src/Containers/AuthoringInterfaceContainer.js]: Remove this

File: squealy/urls.py [hunk 14]
```diff
@@ -3,7 +3,10 @@
 
 import views
 
+
 urlpatterns = [
+   url(r'^swagger-api/$',views.swagger_json_api),
```
Comment (swapnil95) [inline squealy/urls.py]: This has to be swagger.json instead of swagger-api

**Pull request merged**